server/src/main/java/org/opensearch/action/search/DeletePitRequest.java

```diff
@@ -0,0 +1,136 @@
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.opensearch.action.ValidateActions.addValidationError;
+
+/**
+ * Request to delete one or more PIT contexts based on IDs.
+ */
+public class DeletePitRequest extends ActionRequest implements ToXContentObject {
+
+    /**
+     * List of PIT IDs to be deleted , and use "_all" to delete all PIT reader contexts
+     */
+    private List<String> pitIds;
+
+    public DeletePitRequest(StreamInput in) throws IOException {
+        super(in);
+        pitIds = Arrays.asList(in.readStringArray());
+    }
+
+    public DeletePitRequest(String... pitIds) {
+        if (pitIds != null) {
+            this.pitIds = Arrays.asList(pitIds);
+        }
+    }
+
+    public DeletePitRequest(List<String> pitIds) {
+        if (pitIds != null) {
+            this.pitIds = pitIds;
+        }
+    }
+
+    public DeletePitRequest() {}
+
+    public List<String> getPitIds() {
+        return pitIds;
+    }
+
+    public void setPitIds(List<String> pitIds) {
+        this.pitIds = pitIds;
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        ActionRequestValidationException validationException = null;
+        if (pitIds == null || pitIds.isEmpty()) {
+            validationException = addValidationError("no pit ids specified", validationException);
+        }
+        return validationException;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        super.writeTo(out);
+        if (pitIds == null) {
+            out.writeVInt(0);
+        } else {
+            out.writeStringArray(pitIds.toArray(new String[pitIds.size()]));
+        }
+    }
+
+    public void addPitId(String pitId) {
```

Comment:
Do you need to expose this as a public method? I'd consider simplifying this class by:

- remove this method
- make `pitIds` final, and assign it to an empty list in the default constructor case
- in `fromXContent` clear `pidIds` and then add to it directly instead of calling this method